src/Connections/FeedConnection.ts

```diff
@@ -28,6 +33,65 @@ export class FeedConnection extends BaseConnection implements IConnection {
         return new FeedConnection(roomId, event.stateKey, event.content, config.feeds, as, storage);
     }
 
+    static async validateUrl(url: string): Promise<void> {
+        try {
+            new URL(url);
+            const res = await axios.get(url);
+            const contentType = res.headers['content-type'];
+            // we're deliberately liberal here, since different things pop up in the wild
+            if (!contentType.match(/xml/)) {
+                throw new Error(`${contentType} doesn't look like an RSS/Atom feed`);
+            }
+        } catch (err) {
+            throw new Error(`${url} doesn't look like a valid feed URL: ${err}`);
+        }
+    }
+
+    static async provisionConnection(roomId: string, _userId: string, data: Record<string, unknown> = {}, {as, config, storage}: ProvisionConnectionOpts) {
+        if (!config.feeds?.enabled) {
+            throw Error('RSS/Atom feeds are not configured');
+        }
+
+        const url = data.url;
+        if (typeof url !== 'string') {
+            throw new Error('No URL specified');
+        }
+        await FeedConnection.validateUrl(url);
```

Comment:
Arguably we should catch this and wrap it in a nice error, for API purposes.

Reply:
Agreed – done in feeb8891c89d7c83847fb6b5264df90df9aa7d86